.github/workflows/generate.yaml

```diff
@@ -0,0 +1,114 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Generate Library
+on:
+  schedule:
+    # Run at the end of every day. (10:10pm PT, 5:10am UTC)
+    - cron: 10 5 * * *
+  workflow_dispatch:
+
+jobs:
+  run-generator:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      issues: write
+      pull-requests: write
+    concurrency:
+      group: ${{ github.workflow }}
+      cancel-in-progress: true
+    env:
+      LIBRARY_CHECKOUT_PATH: library
+      DATA_SOURCE_CHECKOUT_PATH: google-cloudevents
+
+    steps:
+      - name: Python Library > Checkout Repository
+        uses: actions/checkout@v3
+        with:
+          path: ${{ env.LIBRARY_CHECKOUT_PATH }}
+
+      - name: Proto Schemas > Checkout Repository
+        uses: actions/checkout@v3
+        with:
+          repository: googleapis/google-cloudevents
+          path: ${{ env.DATA_SOURCE_CHECKOUT_PATH }}
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+          cache: 'pip'
+
+      - name: Install Generator Dependencies
+        run: pip install -r ./scripts/requirements.txt
+
+      - name: Run the generator
```

Comment:
Does the generator need to know where the parent directory is through env var?

Reply:
The generator fetches its own copy of the `google-cloudevents` repo, so we actually dont need the Proto Schemas checkout step.